### ISSUE TITLE
Model "saving" event not fired when adding a child object to a OneToMany relationship

### ISSUE THREAD

**Issue by leoquijano** [laravel/framework#21260]
Hello,

I'm using Laravel Framework v5.5.7 on a PHP v7.0. I noticed that when you're adding a child object to a HasToMany relationship, and you instruct Laravel to automatically "touch" the parent object when that happens, the ```saved``` event gets fired, but not the ```saving``` event.

Is this by design?
Suppose we have the following classes:

*Post*
```php
namespace App;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Post extends Model {

  /**
   * Get the comments for this post.
   * @return HasMany
   */
  public function comments() {
    return $this->hasMany('App\Comment', 'post_id');
  }

  public static function boot() {
    parent::boot();
    self::saving(function ($model) {
      \Log::debug("Post - Saving");
    });
    self::saved(function ($model) {
      \Log::debug("Post - Saved");
    });
  }
```

*Comment*
```php
namespace App;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Comment extends Model {

  /**
   * All of the relationships to be touched. 
   *
   * @var array
   */
  protected $touches = ['post'];

  /**
   * Returns the post.
   * @return BelongsTo
   */
  public function post() {
    return $this->belongsTo('App\Post', 'post_id');
  }

  public static function boot() {
    parent::boot();
    self::saving(function($model) {
      \Log::debug("Comment - Saving");
    });
    self::saved(function($model) {
      \Log::debug("Comment - Saved");
    });
  }
}
```

Now, suppose we have the following simple code. It can be in a Command object, or in a web application. The database doesn't matter (the issue shows up in MySQL and in SQLite):

```php
    $post          = new Post();
    $post->name    = "Test Post";
    $post->content = "This is a test post";
    $post->save();

    $comment          = new Comment();
    $comment->subject = "Test Comment";
    $comment->content = "This is a test comment";
    $post->comments()->save($comment);

    sleep(5);
    $post->touch();
```

This is what I get in the log:

```
[2017-09-19 21:25:13] local.DEBUG: Post - Saving  
[2017-09-19 21:25:13] local.DEBUG: Post - Saved  
[2017-09-19 21:25:13] local.DEBUG: Comment - Saving  
[2017-09-19 21:25:13] local.DEBUG: Comment - Saved  
[2017-09-19 21:25:13] local.DEBUG: Post - Saved  
   // ... 5 second delay ...
[2017-09-19 21:25:44] local.DEBUG: Post - Saving  
[2017-09-19 21:25:44] local.DEBUG: Post - Saved  
```

So, basically the "saving" event is fired when:
- You modify the Post object directly.
- You call ```touch``` on the parent, directly.

However, it's not being fired when you add a comment through the OneToMany() relationship. This is a bit inconsistent, since calling ```touch``` directly does fire the event.

There are use cases to firing the ```saving``` event when adding child elements. In my case, I am calculating a parent column's value based on the contents of the child elements. This could also be done in the ```saved``` event, but that would add an extra database query. And also, there's also the use case where you would like to prevent the ```save``` method from running in some cases.

Any ideas of why this could be happening?

**Comment by devcircus** [comment]
Seems to be by design, see below. However maybe it's something that can be considered.

```php
// Illuminate\Database\Eloquent\Concerns\HasRelationships
public function touchOwners()
{
    foreach ($this->touches as $relation) {
        $this->$relation()->touch();
         if ($this->$relation instanceof self) {
            $this->$relation->fireModelEvent('saved', false);
            $this->$relation->touchOwners();
        } elseif ($this->$relation instanceof Collection) {
            $this->$relation->each(function (Model $relation) {
                $relation->touchOwners();
            });
        }
    }
}
```

**Comment by Dylan-DPC** [comment]
Yeah i agree. This isnt a bug. However you can make a proposal in the laravel/internals repo

**Comment by leoquijano** [comment]
Thanks,

Any ideas why it would be designed this way? It seems a bit strange that the ```saving``` event is being fired when you touch the object directly, but not when you touch it through the children elements.

**Comment by devcircus** [comment]
See https://github.com/laravel/framework/pull/11400

Basically someone was wondering the same thing about the saved event and made a PR to add it. Which is why I think they may consider this.

**Comment by leoquijano** [comment]
Ah, I see! 

Ok it probably will be a good idea to make a PR for this one. Is this something Laravel would be willing to accept?

**Comment by Dylan-DPC** [comment]
Yes @leoquijano go ahead and make a PR.

**Comment by leoquijano** [comment]
Hi guys, why was the issue closed? I'm planning to make the PR soon.

**Comment by themsaid** [comment]
@leoquijano it's not really a bug, so a PR with your suggested changes is a better place for discussion since we can look into an implementation.

**Comment by andreladocruz** [comment]
@Dylan-DPC or @themsaid , do you know if a PR were made for this? 

How can I change this behavior in my project?
